test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -3364,14 +3369,16 @@ public async Task MaxOrchestrationAction_MaxReached_OrchestrationFails()
             {
                 await host.StartAsync();
 
-                var client = await host.StartOrchestratorAsync(nameof(TestOrchestrations.TwoOrchestratorActivityActions), "", this.output);
+                var counterEntityId = new EntityId("Counter", Guid.NewGuid().ToString());
+
+                var client = await host.StartOrchestratorAsync(orchestratorFunctionNames[0], counterEntityId, this.output);
                 DurableOrchestrationStatus status = await client.WaitForCompletionAsync(this.output);
 
                 Assert.NotNull(status);
                 Assert.Equal(OrchestrationRuntimeStatus.Failed, status.RuntimeStatus);
                 Assert.Equal(

```

Comment:
It seems like this test would pass even if we accidentally threw the exception sometime before the 7th action. To make the test more precise, I suggest you update the orchestration with a `ctx.SetCustomStatus(...)` call right before the expected failure. You can then add an `Assert` here which checks for that custom status in the `status.CustomStatus` field.

Reply:
That assert statement would pass, but the test would only pass if the next assert statement, that checks the error message, passes as well. But I've never used SetCustomStatus before and didn't think about that when I wrote this test. I will add that.

Reply:
Actually, is this necessary? It doesn't change the test from relying on the last assert to compare the output to the error message.

Reply:
You're right. I think the actual (potential) problem I had in mind was slightly different. My worry was actually more that you won't know if the exception is thrown too early. For example, let's say you added one more API call to your orchestration but forgot to update `maxActions` from 7 to 8. The test will still pass, but your last API call will not have been invoked. You could make that kind of problem less likely by checking a custom status which is set only after all the API calls you expected to run have actually run. It's a minor thing, but I think it will make the test a bit more robust.